src/components/DashboardPlots/BoxChart/sections/BoxChartControls.tsx

```diff
@@ -0,0 +1,175 @@
+import { useTranslation } from 'react-i18next';
+import { GxSelect } from '../../../../shared/components/GxSelect';
+import { BoxChartControlsProps, BoxChartValueType, BoxChartHueValueOptions } from './BoxChartControls.types';
+
+const AVAILABLE_HUE_OPTIONS: BoxChartHueValueOptions[] = ['none', 'clusterId', 'roi'];
+
+export const BoxChartControls = ({
+  selectedValue,
+  selectedROIs,
+  selectedHue,
+  selectedValueType,
+  onRoiChange,
+  onValueChange,
+  onHueChange,
+  onValueTypeChange
+}: BoxChartControlsProps) => {
+  const { t } = useTranslation();
+  const { segmentationMetadata, selectedCells } = useCellSegmentationLayerStore();
+
+  const availableROIOptions = useMemo(
+    () =>
+      selectedCells
+        .map((feature) => {
+          const polygonId = feature.roiId;
+          if (polygonId === undefined) return null;
+          return {
+            value: String(polygonId),
+            label: t('general.roiEntry', { index: polygonId })
+          };
+        })
+        .filter((entry) => !!entry),
+    [t, selectedCells]
+  );
+
+  useEffect(() => {
+    const availableId = availableROIOptions.map((entry) => Number(entry.value));
+    onRoiChange(selectedROIs.filter((roiId) => availableId.includes(roiId)));
+    //eslint-disable-next-line
+  }, [availableROIOptions]);
+
+  const availableGenes = segmentationMetadata?.geneNames || [];
+  const availableProteins = segmentationMetadata?.proteinNames || [];
+
+  const handleRoiChange = (event: SelectChangeEvent<string[] | string>) => {
+    const {
+      target: { value }
+    } = event;
+    onRoiChange(Array.isArray(value) ? value.map(Number) : [Number(value)]);
+  };
+
+  return (
+    <Box sx={sx.container}>
+      <FormControl
+        sx={sx.controlWrapper}
+        size="small"
+      >
+        <Typography sx={sx.inputLabel}>{t('dashboard.availableROILabel')}:</Typography>
+        <GxMultiSelect
+          value={selectedROIs.map(String)}
+          onChange={handleRoiChange}
+          options={availableROIOptions}
+          renderValue={(selected) => {
+            if (!selected.length) {
+              return `${t('dashboard.selectROILabel')}...`;
+            } else if (selected.length === availableROIOptions.length) {
+              return t('dashboard.allROILabel');
+            } else if (selected.length < 3) {
+              return selected.map((entry) => t('general.roiEntry', { index: entry })).join(', ');
+            }
+
+            return t('dashboard.multipleROISelected', { count: selected.length });
+          }}
+        />
+      </FormControl>
+      <FormControl
+        sx={sx.controlWrapper}
+        size="small"
+      >
+        <Typography sx={sx.inputLabel}>{t('dashboard.plotValueType')}:</Typography>
+        <GxSelect
+          value={selectedValueType}
+          fullWidth
+          MenuProps={{
+            sx: {
+              maxHeight: '500px'
+            }
+          }}
```

Comment:
I see a few repetitions of `maxHeight: '500px'`. Could you move it to a single place and reuse it?